aws/resource_aws_ssm_resource_data_sync.go

```diff
@@ -0,0 +1,180 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSsmResourceDataSync() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSsmResourceDataSyncCreate,
+		Read:   resourceAwsSsmResourceDataSyncRead,
+		Delete: resourceAwsSsmResourceDataSyncDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"destination": {
+				Type:     schema.TypeSet,
```

Comment:
Since this is a singleton I think we can just make it `TypeList` as there's no value in computing the hash and it would also complicate referencing. TypeList can be referenced as `s3_destination.0.key` etc., whereas TypeSet is non-trivial to reference because the address is `s3_destination.<computed-idx>.key`.